samples/dnn/object_detection.cpp

```diff
@@ -104,44 +137,135 @@ int main(int argc, char** argv)
     else
         cap.open(parser.get<int>("device"));
 
-    // Process frames.
-    Mat frame, blob;
-    while (waitKey(1) < 0)
-    {
-        cap >> frame;
-        if (frame.empty())
+    bool process = true;
+
+    // Frames capturing thread
+    std::mutex framesQueueMutex;
+    QueueFPS<Mat> framesQueue;
+    std::thread framesThread([&](){
+        Mat frame;
+        while (process)
         {
-            waitKey();
-            break;
+            cap >> frame;
+            if (!frame.empty())
+            {
+                framesQueueMutex.lock();
+                framesQueue.push(frame.clone());
+                framesQueueMutex.unlock();
+            }
+            else
+                break;
         }
+    });
+
+    // Frames processing thread
+    std::mutex predictionsMutex;
+    std::queue<Mat> processedFramesQueue;
+    QueueFPS<std::vector<Mat> > predictionsQueue;
+    std::thread processingThread([&](){
+        std::queue<std::future<Mat> > futureOutputs;
+        Mat blob;
+        while (process)
+        {
+            // Get a next frame
+            Mat frame;
+            framesQueueMutex.lock();
```

Comment:
Use scope-based `std::shared_lock` or `cv::AutoLock` (with `cv::Mutex`).

Reply:
Replaced to `std::lock_guard`, thanks!